src/Renderer/LayeredMaterial.js

```diff
@@ -244,83 +283,87 @@ LayeredMaterial.prototype.setTexture = function(texture, layer, slot, pitScale)
 };
 
-LayeredMaterial.prototype.setParam = function(param) {
-    this.uniforms.nColorLayer.value = param.length;
-    for (var l = 0; l < param.length; l++) {
-        this.layerIdToIndex[param[l].idLayer] = l;
-        this.paramLayers[l] = new THREE.Vector4(
-            param[l].layerTexturesOffset,
-            param[l].tileMT === 'PM' ? 1 : 0, param[l].visible, param[l].opacity);
-        this.paramBLayers[l] = new THREE.Vector2(param[l].fx, 0.0);
-    }
+LayeredMaterial.prototype.getNbTexturesByIndex = function(index) {
+    return this.nbTexturesLayers[index];
 };
 
 LayeredMaterial.prototype.getLayerTextureOffset = function(layer) {
-    var index = this.layerIdToIndex[layer];
-    if (index !== undefined) {
-        return this.paramLayers[index].x;
-    } else {
-        return -1;
-    }
+    var index = this.layers.indexOf(layer);
+    return index > -1 ? this.getOffsetByIndex(index) : -1;
 };
 
-LayeredMaterial.prototype.setTexturesLayer = function(textures, layer, slotOffset) {
-    for (var i = 0, max = textures.length; i < max; i++) {
-        if (textures[i]) {
-            this.setTexture(textures[i].texture, layer, i + (slotOffset || 0), textures[i].pitch);
-        }
-    }
+LayeredMaterial.prototype.setLightingOn = function(enable) {
+    this.uniforms.lightingOn.value = enable;
 };
 
-LayeredMaterial.prototype.setLightingOn = function(enable) {
-    this.uniforms.lightingOn.value = enable === true ? 1 : 0;
+LayeredMaterial.prototype.setLayerFx = function(id, fx) {
+        this.uniforms.paramLayers.value[id].z = fx;
 };
 
-LayeredMaterial.prototype.setLayerOpacity = function(id, opacity) {
+LayeredMaterial.prototype.setLayerOffset = function(id, offset) {
+        this.uniforms.paramLayers.value[id].x = offset;
+};
 
-    if (this.paramLayers[id]) {
-        this.paramLayers[id].w = opacity;
-    }
+LayeredMaterial.prototype.setLayerUV = function(id, idUV) {
+        this.uniforms.paramLayers.value[id].y = idUV;
+};
+
+LayeredMaterial.prototype.setLayerOpacity = function(id, opacity) {
+    if(this.uniforms.paramLayers.value[id])
+        this.uniforms.paramLayers.value[id].w = opacity;
 };
 
 LayeredMaterial.prototype.setLayerVisibility = function(id, visible) {
+        this.uniforms.visibility.value[id] = visible;
+};
 
-    if (this.paramLayers[id]) {
-        this.paramLayers[id].z = visible ? 1 : 0;
-    }
+LayeredMaterial.prototype.setLayerNbTextures = function(id, nb) {
+        this.nbTexturesLayers[id] = nb;
+};
+
+LayeredMaterial.prototype.nbLoadedTextures = function() {
+    return this.nbTextures[ELEVATION] + this.nbTextures[COLOR];
 };
 
-LayeredMaterial.prototype.setNbLayersColor = function(n) {
-    this.uniforms.nColorLayer.value = n;
+LayeredMaterial.prototype.getNumberColorLayer = function() {
+    return this.layers.length;
 };
 
-LayeredMaterial.prototype.getLevelLayerColor = function(id, slot) {
+LayeredMaterial.prototype.getLevelLayerColor = function(id, layer) {
```

Comment:
This function was badely named. And so are the args.

WDYT of: `getColorLayerLevel = function(layerType, colorLayerId)`?